src/mym.cpp

```diff
@@ -888,6 +894,120 @@ void mexFunction(int nlhs, mxArray*plhs[], int nrhs, const mxArray*prhs[]) {
         mxFree(i_pr);
         mysql_free_result(res);
     }
+    else if (q==SERIALIZE) {
+        // serialize the matlab variable, it can be an array, cell, struct etc
+        //******************PLACEHOLDER PROCESSING******************
+        // global placeholders variables and constant
+        const unsigned nex = nrhs-jarg-1;   // expected number of placeholders
+        unsigned query_flags = 0;
+        unsigned nb_flags = 0;
+        unsigned nac = 0;                                   // actual number
+        size_t lq = strlen(query);  // original query length, needed at some point
+
+        if (nex) {
+            // local placeholders variables and constant
+            char** po = 0;                              // pointer to placeholders openning symbols
+            char** pc = 0;                              // pointer to placeholders closing symbols
+            bool*  pec = 0;                             // pointer to 'enable compression field'
+            char** pa = 0;                              // pointer to placeholder in-string argument
+            unsigned* ps = 0;                           // pointer to placeholders size (in bytes)
+            pfserial* pf = 0;                           // pointer to serialization function
+            // LOOK FOR THE PLACEHOLDERS
+            po = (char**)mxCalloc(nex+1, sizeof(char*));
+            pc = (char**)mxCalloc(nex+1, sizeof(char*));
+            if ((po[nac++] = strstr(query, PH_OPEN)))

```

Comment:
Let's take the assignment out of the the if condition for better readability.

Reply:
Dimitri, this is copy from the original code, I am not sure if it is worthwhile to spend the time to just clean it in the section we have added